### PR TITLE
NetworkMonitor - Use portal

### LINKED ISSUE #28: Use NetworkMonitor portal for Aura.NetworkMonitor
https://flatpak.github.io/xdg-desktop-portal/#gdbus-method-org-freedesktop-portal-NetworkMonitor.GetAvailable

This way we will not need to have NetworkManager DBus permission in flatpak.

### PR DESCRIPTION
**Pull request by fsobolev**
Closes #28

### CODE REVIEW

File: Nickvision.Aura/Network/INetworkMonitor.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    Task<NetworkMonitorProperties> GetAllAsync();
+    Task SetAsync(string prop, object val);
+    Task<IDisposable> WatchPropertiesAsync(Action<PropertyChanges> handler);
+}
+
+[Dictionary]
+internal class NetworkMonitorProperties
+{
+    private uint _version = default(uint);
+    public uint Version
```
Comment (nlogozzo) [inline Nickvision.Aura/Network/INetworkMonitor.cs:28]: Add whitespace

File: Nickvision.Aura/Network/INetworkMonitor.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    {
+        get
+        {
+            return _version;
+        }
+
+        set
+        {
+            _version = (value);
```
Comment (nlogozzo) [inline Nickvision.Aura/Network/INetworkMonitor.cs:37]: Remove the `()` around value

File: Nickvision.Aura/Network/INetworkMonitor.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        }
+    }
+}
+
+internal static class NetworkMonitorExtensions
+{
+    public static Task<uint> GetVersionAsync(this INetworkMonitor o) => o.GetAsync<uint>("version");
```
Comment (nlogozzo) [inline Nickvision.Aura/Network/INetworkMonitor.cs:44]:
```csharp
public static async Task<uint> GetVersionAsync(this INetworkMonitor o) => await o.GetAsync<uint>("version");
```

**Review by nlogozzo — CHANGES_REQUESTED** [review]

**Comment by nlogozzo** [comment]
If you could also merge from main and then update the `PackageReleaseNotes` in csproj.

File: Nickvision.Aura/Network/INetworkMonitor.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+[DBusInterface("org.freedesktop.portal.NetworkMonitor")]
+internal interface INetworkMonitor : IDBusObject
+{
+    Task<bool> GetAvailableAsync();
+    Task<bool> GetMeteredAsync();
+    Task<uint> GetConnectivityAsync();
+    Task<IDictionary<string, object>> GetStatusAsync();
+    Task<bool> CanReachAsync(string Hostname, uint Port);
+    Task<IDisposable> WatchchangedAsync(Action handler);
```
Comment (nlogozzo) [inline Nickvision.Aura/Network/INetworkMonitor.cs:17]: Should this be `WatchchangedAsync`? Or `WatchChangedAsync`?

**Review by nlogozzo — CHANGES_REQUESTED** [review]

File: Nickvision.Aura/Network/INetworkMonitor.cs (hunk 4 above)
Comment (fsobolev) [reply Nickvision.Aura/Network/INetworkMonitor.cs:17]: The signal name is `changed` (lowercase), so generated method is `WatchchangedAsync` (I know, doesn't look right)
https://flatpak.github.io/xdg-desktop-portal/#gdbus-signal-org-freedesktop-portal-NetworkMonitor.changed

**Comment by nlogozzo** [comment]
Give me 10 minutes and I'll release V2023.9.5 to nuget :)

File: Nickvision.Aura/Network/INetworkMonitor.cs (hunk 4 above)
Comment (fsobolev) [reply Nickvision.Aura/Network/INetworkMonitor.cs:17]: I just tried to make it `WatchChangedAsync` and it stopped working.

**Pull request merged**